src/routers/user/index.ts

```diff
@@ -192,4 +197,37 @@ router.delete('/:id', auth, async (req: Request, res: Response) => {
   }
 });
 
+// check membership by email or watiam user id
+router.post(
+  '/membership-check',
+  routeValidator('/membership-check'),
+  validate,
+  async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const { emailOrWatIAMUserId } = req.body as MembershipCheckRequestBody;
+
+      const isEmail = validator.isEmail(emailOrWatIAMUserId);
+      const conditions: Record<string, string> = {};
```

Comment:
I think we can query for the user with `emailOrWatIAMUserId` not so?

Reply:
Not sure I understand. I'm using `emailOrWatIAMUserId` and query by either `email`  or `watIAMUserId` depending on if it's an email or not.

Do you mean like doing an `$or` and querying over both fields at the same time?

Reply:
My bad I'm explaining horribly.

I meant instead of `const user = await UserModel.findOne(conditions);`

You could do `const user = await UserModel.findOne(emailOrWatIAMUserId);` right?

Reply:
So you only have to check whether it was an email when you're responding to the request

Reply:
Oh wait nevermind. I'm talking str8 🗑. You have to specify the field🤣
Code looks great bro